### PR TITLE
Tool 1126 swiftpm

### PR DESCRIPTION
**Pull request by lpusok**

### CODE REVIEW

**Comment by lpusok** [comment]
@trapacska Please add pullassigner.

File: build.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+		log.RWarnf("xcode-archive", "swift-packages-cache-invalid", nil, "swift packages cache is in an invalid state")
+		if err := os.RemoveAll(swiftPackagesPath); err != nil {
+			return output, fmt.Errorf("failed to remove invalid Swift package caches, error: %s", err)
+		}
+		return runBuildCommand(buildCmd, useXcpretty)
+	}
+	return output, err
+}
+
+func runBuildCommand(buildCmd *xcodebuild.CommandBuilder, useXcpretty bool) (string, error) {
```
Comment (trapacska) [inline build.go]: this name is very not describing the logic of the function

Comment (trapacska) [reply build.go]: unsure if this logic is need to be seperated

File: build.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+	buildRootCmd := xcodeCmd.Command()
+	buildRootCmd.SetStdout(io.MultiWriter(os.Stdout, output))
+	buildRootCmd.SetStderr(io.MultiWriter(os.Stderr, output))
+	return buildRootCmd, nil
+}
+
+func runCommand(buildCmd *xcodebuild.CommandBuilder, useXcpretty bool) (string, error) {
+	var output bytes.Buffer
+	xcodebuildCmd, xcprettyCmd := prepareCommand(buildCmd, useXcpretty, &output)
```
Comment (trapacska) [inline build.go]: buildCmd.PrintableCmd() and xcprettyCmd.Command().PrintableCmd() is required as it is differs in this case

File: bitrise.yml [hunk 3]
```diff
@@ -23,11 +23,11 @@ workflows:
     - test_workspace
     - test_multi_target_workspace_custom_derived_data_path
     - test_multi_target_workspace
-    - test_react_native
-    - test_react_native_workspace
-    - test_react_native_custom_target_build_dir
+    # Disabled react-ntaive due ti detox installation failure
```
[the remark this answers is not in the document]
Comment (lpusok) [reply bitrise.yml]: yes.

File: build.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
```
Comment (trapacska) [inline build.go]: xcodebuildCmd, xcprettyCmd := prepareCommand(buildCmd, useXcpretty, &output)

cmd := xcodeBuildCmd
if xcprettyCmd != nil {
cmd = xcprettyCmd.Command()
}

cmd.PrintableArgs...

**Comment by lpusok** [comment]
@trapacska please merge.

File: main.go [hunk 5]
```diff
@@ -249,6 +248,13 @@ The log file is stored in $BITRISE_DEPLOY_DIR, and its full path is available in
 			}
 			failf("Build failed, error: %s", err)
 		}
+
+		// Cache swift PM
```
Comment (trapacska) [inline main.go]: I think caching always should be at the very end of the logic, to avoid adding anything to a cache even if the steps had a failure

Comment (lpusok) [reply main.go]: fixed.

**Pull request merged**